### PR TITLE
Stats: Adding an icon to indicate titles with links

### PR DESCRIPTION
**Pull request by grzegorz-cp**
#### Proposed Changes

* adding a chevron icon with hover transformation to indicate titles that are also links (applicable to a highlights component and modules with horizontal bars)

#### Testing Instructions

* open `Stats` page and verify that components with horizontal bars on the `Traffic` page have chevron icons and are links redirecting to the details page
* navigate to `Insights` page and verify that `2022 in review` highlight component also has a chevron and link to more details

Before:
Screenshot 2022-12-08 at 10 25 15 AM

Screenshot 2022-12-08 at 10 25 37 AM

After:
Screenshot 2022-12-08 at 10 24 35 AM

Screenshot 2022-12-08 at 10 25 52 AM

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #70849

### CODE REVIEW

File: packages/components/src/highlight-cards/annual-highlight-cards.tsx [hunk 1]
```diff
@@ -27,9 +35,12 @@ export default function AnnualHighlightCards( {
 
 	const header = (
 		<h1 className="highlight-cards-heading">
-			{ Number.isFinite( year )
-				? translate( '%(year)s in review', { args: { year } } )
-				: translate( 'Year in review' ) }
+			<span>
+				{ Number.isFinite( year )
```
Comment (kangzj) [inline packages/components/src/highlight-cards/annual-highlight-cards.tsx:39]: Not related to the PR, but `Number.isFinite( year )`  doesn't look right, does it?

**Review by kangzj — APPROVED** [review]
This is a really nice improvement! LGTM

----UPDATE----
It would be nice if the icon would be of the same color as the text on hovering.

**Comment by grzegorz-cp** [comment]
> It would be nice if the icon would be of the same color as the text on hovering.

Good catch. It's updated now.

File: packages/components/src/highlight-cards/annual-highlight-cards.tsx (hunk 1 above)
Comment (grzegorz-cp) [reply packages/components/src/highlight-cards/annual-highlight-cards.tsx:39]: I think it was added to ensure that we deal with a number or a proper value after some calculations. Would you use something else?

**Review by jsnmoon — CHANGES_REQUESTED** [review]
Two things to address:

image

- [ ] Spacing and alignment are now broken for the 7-day highlights header.

image

- [ ] The link chevron is ~two pixels higher than the baseline. This appears to be fixed if `align-items: flex-end;` is used instead of `align-items: center;`.

Everything else looks solid.

File: packages/components/src/highlight-cards/annual-highlight-cards.tsx (hunk 1 above)
Comment (jsnmoon) [reply packages/components/src/highlight-cards/annual-highlight-cards.tsx:39]: IIRC it was to check whether `year` was a valid number.

**Comment by grzegorz-cp** [comment]
Thanks for the feedback. I used `flex-end` in the beginning, but this didn't look right to me for the highlight section, so I converted everything to use the centred version. I can keep `center` for the highlights and `flex-end` for the modules.

**Comment by jeffgolenski** [comment]
Ah this is my bad for not updating a proper direction with the design. Last week, @keoshi and I kept exploring a few patterns for this!

1) For the 2022 in review, let's keep the same text pattern that we're using on the traffic page and simply add this text and turn it into a link. This way, each area is consistent. The chevrons as links aren't really used on wordpress.com outside of the old stats that I've seen. 

Screen Shot 2022-12-07 at 6 47 59 PM

2) Card titles shouldn't be links anymore. Again—apologies if we didn't communicate that properly early on! Each individual card should have a "view all" link at the bottom and that'll serve as the one and only link to the detailed sections. This way we can reduce some code and also the visual clutter in these cards

Screen Shot 2022-12-07 at 6 51 55 PM

**Comment by grzegorz-cp** [comment]
I'll open a new PR with the updated scope.

**Pull request closed without merging**